src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -564,6 +593,47 @@ private static Array BindArray(Type type, IEnumerable? source, IConfiguration co
             return result;
         }
 
+        [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the Array so its members may be trimmed.")]
+        private static object BindSet(Type type, IEnumerable? source, IConfiguration config, BinderOptions options)
+        {
+            Type elementType = type.GetGenericArguments()[0];
+
+            Type genericType = typeof(HashSet<>).MakeGenericType(type.GenericTypeArguments[0]);
+            object instance = Activator.CreateInstance(genericType)!;
+
+            MethodInfo addMethod = genericType.GetMethod("Add", DeclaredOnlyLookup)!;
+
+            if (source != null)
+            {
+                foreach (object? item in source)
+                {
+                    addMethod.Invoke(instance, new[] {item});
```

Comment:
>new[]  [](http://example.com/codeflow?start=47&length=6)

we always passing the arguments in array of one item. can we just allocate the array once and use it for all calls? This will save many allocations I guess. We can use this array down in this method too.